libs/base/src/utils/COutputLogger.cpp

```diff
@@ -65,49 +74,101 @@ void COutputLogger::log(const std::string& msg_str, const VerbosityLevel& level)
 
 	m_curr_level = global_level;
 }
-
-void COutputLogger::log(const std::string& msg_str) const {
+void COutputLogger::log(const std::string& msg_str, const VerbosityLevel& level) const {
 	warnForLogConstMethod();
 
 	TMsg msg(msg_str, *this);
+	msg.level = level;
 	msg.dumpToConsole();
 }
 
-void COutputLogger::log(const std::string& msg_str, const VerbosityLevel& level) const {
-	warnForLogConstMethod();
+void COutputLogger::logFmt(const char* fmt, ...) {
+	// see MRPT/libs/base/src/utils/CDeugOutputCapable.cpp for the iniitial
+	// implementtion
 
-	TMsg msg(msg_str, *this);
-	msg.level = level;
-	msg.dumpToConsole();
+	// check for NULL pointer
+	if (!fmt) return;
+
+	// initialize the va_list and let generateStringFromFormat do the work
+	// http://c-faq.com/varargs/handoff.html
+	va_list argp;
+	va_start(argp, fmt);
+	std::string str = this->generateStringFromFormat(fmt, argp);
+	va_end(argp);
+
+	this->log(str);
 }
 
-void COutputLogger::warnForLogConstMethod() const {
-	if (m_times_for_log_const++ >= m_times_for_log_const_thresh) {
-		return;
-	}
+void COutputLogger::logFmt(const char* fmt, ...) const {
+	if (!fmt) return;
 
-	// warn the user of the method behavior
-	std::string msg_str("Using the log method inside a const function/method. Message will not be recorded in COutputLogger history");
-	TMsg warning_msg(msg_str, *this);
-	warning_msg.level = LVL_WARN;
-	warning_msg.dumpToConsole();
-	mrpt::system::sleep(1000);
+	va_list argp;
+	va_start(argp, fmt);
+	std::string str = this->generateStringFromFormat(fmt, argp);
+	va_end(argp);
+
+	this->log(str);
 }
 
+std::string COutputLogger::generateStringFromFormat(const char* fmt, va_list argp) const{
```

Comment:
Have a common generateStringFromFormat function to be called from both the logFmt **and** logFmt const